scratchpad/bench/matmul.py

```diff
@@ -20,159 +20,34 @@
+    @classmethod
+    def get_params(cls, *sizes):
         ops = np.prod(sizes[0]) * sizes[1][1]
         param_in = np.prod(sizes[0]) + np.prod(sizes[1])
         param_out = sizes[0][0]*sizes[1][1]
         return ops.item(), param_in.item(), param_out
 
 
-
-class Numpy(Matmul):
-    @staticmethod
-    def get_dtype(dtype):
-        return {
-            'float':np.float32
-            ,'double': np.float64
-            ,'complex64': np.complex64
-            ,'complex128': np.complex128
-        }[dtype]
-
-    @classmethod
-    def gen_tensor(cls, *sizes, dtype='float'):
-        dtype_t = cls.get_dtype(dtype)
-        return np.random.rand(*sizes).astype(dtype_t)
-
-    @staticmethod
-    def get_operation():
-        return np.matmul
-
-
-class Torch(Matmul):
-    torch.backends.cuda.matmul.allow_tf32 = False
-    gpu_tensor = ['float', 'double']
-
-    @staticmethod
-    def get_dtype(dtype):
-        return {
-            'float':torch.cuda.FloatTensor
-            ,'double': torch.cuda.DoubleTensor
-            ,'complex64': torch.complex64
-            ,'complex128': torch.complex128
-        }[dtype]
-
-    @classmethod
-    def gen_tensor(cls, *sizes, dtype='float'):
-        if dtype in cls.gpu_tensor:
-            dtype = cls.get_dtype(dtype)
-            return dtype(*sizes).normal_()
-        else:
-            dtype = cls.get_dtype(dtype)
-            return torch.rand(*sizes, dtype=dtype, device='cuda')
-
-    @staticmethod
-    def get_operation():
-        return torch.matmul
-
-
-class TorchCuda(Torch):
-    @classmethod
-    @contextmanager
-    def timing(cls, **kwargs):
-        class Foo:
-            pass
-        res = Foo()
-        start = torch.cuda.Event(enable_timing=True)
-        end = torch.cuda.Event(enable_timing=True)
-        start.record()
-        yield res
-        end.record()
-        torch.cuda.synchronize()
-        res.result = start.elapsed_time(end)/1000
-
-
-class Cupy(Matmul):
-    @classmethod
-    @contextmanager
-    def timing(cls, **kwargs):
-        class Foo:
-            pass
-        res = Foo()
-        start = cupy.cuda.Event(disable_timing=False)
-        end = cupy.cuda.Event(disable_timing=False)
-        start.record()
-        yield res
-        end.record()
-        end.synchronize()
-        res.result = cupy.cuda.get_elapsed_time(start, end)/1000
-
-    @staticmethod
-    def get_dtype(dtype):
-        return {
-            'float':cupy.float32
-            ,'double': cupy.float64
-            ,'complex64': cupy.complex64
-            ,'complex128': cupy.complex128
-        }[dtype]
-
-    @classmethod
-    def gen_tensor(cls, *sizes, dtype='float'):
-        dtype_t = cls.get_dtype(dtype)
-        return cupy.random.rand(*sizes).astype(dtype_t)
-
-    @staticmethod
-    def get_operation():
-        with pyrofiler.timing('cblas handler'):
-            cupy.cuda.device.get_cublas_handle()
-        return cupy.matmul
-
-
-class CuTensor(Cupy):
+class CuTensorMatmul(CuTensor):
```

Comment:
Hm, this is the analogy of `BlueCircle` - two dimensions in one class. Is there a reason you picked cutensor matmul to have a dedicated class? Why not `bench = MatmulBench(); bench.benchmark(backend=Cupy())`?